Error compiling for ESP8266 (and possibly many other non-AVR architectures)

Arduino IDE 1.8.5 fails to compile this library for ESP8266 (and presumably a number of other non-AVR architectures, e.g., SAMD and STM32) due to these errors:

```
/Users/Macro/Development/Arduino/Workspace/libraries/SparkFun_VEML6075_Arduino_Library/src/SparkFun_VEML6075_Arduino_Library.cpp: In member function 'VEML6075_error_t VEML6075::begin(TwoWire&)':
/Users/Macro/Development/Arduino/Workspace/libraries/SparkFun_VEML6075_Arduino_Library/src/SparkFun_VEML6075_Arduino_Library.cpp:123:15: error: 'class TwoWire' has no member named 'end'
     _i2cPort->end();
               ^
```

The `Wire.end()` (which is absent on the ESP8266 Arduino core) issue should be a trivial fix, perhaps with an `#ifdef` statement?

```
/Users/Macro/Development/Arduino/Workspace/libraries/SparkFun_VEML6075_Arduino_Library/src/SparkFun_VEML6075_Arduino_Library.cpp: In member function 'VEML6075::veml6075_uv_it_t VEML6075::getIntegrationTime()':
/Users/Macro/Development/Arduino/Workspace/libraries/SparkFun_VEML6075_Arduino_Library/src/SparkFun_VEML6075_Arduino_Library.cpp:217:41: error: invalid conversion from 'int' to 'VEML6075::veml6075_uv_it_t' [-fpermissive]
     return (conf & VEML6075_UV_IT_MASK) >> VEML6075_UV_IT_SHIFT;
                                         ^
/Users/Macro/Development/Arduino/Workspace/libraries/SparkFun_VEML6075_Arduino_Library/src/SparkFun_VEML6075_Arduino_Library.cpp: In member function 'VEML6075::veml6075_hd_t VEML6075::getHighDynamic()':
/Users/Macro/Development/Arduino/Workspace/libraries/SparkFun_VEML6075_Arduino_Library/src/SparkFun_VEML6075_Arduino_Library.cpp:254:38: error: invalid conversion from 'int' to 'VEML6075::veml6075_hd_t' [-fpermissive]
     return (conf & VEML6075_HD_MASK) >> VEML6075_HD_SHIFT;    
                                      ^
/Users/Macro/Development/Arduino/Workspace/libraries/SparkFun_VEML6075_Arduino_Library/src/SparkFun_VEML6075_Arduino_Library.cpp: In member function 'VEML6075::veml6075_uv_trig_t VEML6075::getTrigger()':
/Users/Macro/Development/Arduino/Workspace/libraries/SparkFun_VEML6075_Arduino_Library/src/SparkFun_VEML6075_Arduino_Library.cpp:283:40: error: invalid conversion from 'int' to 'VEML6075::veml6075_uv_trig_t' [-fpermissive]
     return (conf & VEML6075_TRIG_MASK) >> VEML6075_TRIG_SHIFT;    
                                        ^
/Users/Macro/Development/Arduino/Workspace/libraries/SparkFun_VEML6075_Arduino_Library/src/SparkFun_VEML6075_Arduino_Library.cpp: In member function 'VEML6075::veml6075_af_t VEML6075::getAutoForce()':
/Users/Macro/Development/Arduino/Workspace/libraries/SparkFun_VEML6075_Arduino_Library/src/SparkFun_VEML6075_Arduino_Library.cpp:313:38: error: invalid conversion from 'int' to 'VEML6075::veml6075_af_t' [-fpermissive]
     return (conf & VEML6075_AF_MASK) >> VEML6075_AF_SHIFT;        
                                      ^
```

The above conversion errors seem can be fixed by [static casting](https://stackoverflow.com/questions/10593923/converting-from-int-to-enum) like `static_cast<veml6075_uv_it_t>(newValue)`. The error gets eliminated after the change, though I have not tested with the actual device yet.

```
/Users/Macro/Development/Arduino/Workspace/libraries/SparkFun_VEML6075_Arduino_Library/src/SparkFun_VEML6075_Arduino_Library.cpp: At global scope:
/Users/Macro/Development/Arduino/Workspace/libraries/SparkFun_VEML6075_Arduino_Library/src/SparkFun_VEML6075_Arduino_Library.cpp:316:57: error: default argument given for parameter 1 of 'VEML6075_error_t VEML6075::powerOn(boolean)' [-fpermissive]
 VEML6075_error_t VEML6075::powerOn(boolean enable = true)
                                                         ^
In file included from /Users/Macro/Development/Arduino/Workspace/libraries/SparkFun_VEML6075_Arduino_Library/src/SparkFun_VEML6075_Arduino_Library.cpp:26:0:
/Users/Macro/Development/Arduino/Workspace/libraries/SparkFun_VEML6075_Arduino_Library/src/SparkFun_VEML6075_Arduino_Library.h:125:22: error: after previous specification in 'VEML6075_error_t VEML6075::powerOn(boolean)' [-fpermissive]
     VEML6075_error_t powerOn(boolean enable = true);
                      ^
/Users/Macro/Development/Arduino/Workspace/libraries/SparkFun_VEML6075_Arduino_Library/src/SparkFun_VEML6075_Arduino_Library.cpp:321:60: error: default argument given for parameter 1 of 'VEML6075_error_t VEML6075::shutdown(boolean)' [-fpermissive]
 VEML6075_error_t VEML6075::shutdown(boolean shutdown = true)
                                                            ^
In file included from /Users/Macro/Development/Arduino/Workspace/libraries/SparkFun_VEML6075_Arduino_Library/src/SparkFun_VEML6075_Arduino_Library.cpp:26:0:
/Users/Macro/Development/Arduino/Workspace/libraries/SparkFun_VEML6075_Arduino_Library/src/SparkFun_VEML6075_Arduino_Library.h:126:22: error: after previous specification in 'VEML6075_error_t VEML6075::shutdown(boolean)' [-fpermissive]
     VEML6075_error_t shutdown(boolean shutdown = true);
                      ^
exit status 1
Error compiling for board Generic ESP8266 Module.
```

The default arguments are already defined in the header file, hence the errors. Removing the `= true` parts in the `.cpp` file should fix the problem.

--- a/src/SparkFun_VEML6075_Arduino_Library.cpp
+++ b/src/SparkFun_VEML6075_Arduino_Library.cpp
@@ -120,7 +120,9 @@ VEML6075_error_t VEML6075::begin(TwoWire &wirePort)
     _deviceAddress = VEML6075_ADDRESS;
     _i2cPort = &wirePort;
 
-    _i2cPort->end();
+    #ifdef WIRE_HAS_END
+        _i2cPort->end();
+    #endif
     _i2cPort->begin();
 
     err = _connected();
@@ -214,7 +216,7 @@ VEML6075::veml6075_uv_it_t VEML6075::getIntegrationTime(void)
     {
         return IT_INVALID;
     }
-    return (conf & VEML6075_UV_IT_MASK) >> VEML6075_UV_IT_SHIFT;
+    return static_cast<VEML6075::veml6075_uv_it_t>((conf & VEML6075_UV_IT_MASK) >> VEML6075_UV_IT_SHIFT);
 }
 
 VEML6075_error_t VEML6075::setHighDynamic(VEML6075::veml6075_hd_t hd)
@@ -251,7 +253,7 @@ VEML6075::veml6075_hd_t VEML6075::getHighDynamic(void)
     {
         return HD_INVALID;
     }
-    return (conf & VEML6075_HD_MASK) >> VEML6075_HD_SHIFT;    
+    return static_cast<VEML6075::veml6075_hd_t>((conf & VEML6075_HD_MASK) >> VEML6075_HD_SHIFT);
 }
 
 VEML6075_error_t VEML6075::setTrigger(VEML6075::veml6075_uv_trig_t trig)
@@ -280,7 +282,7 @@ VEML6075::veml6075_uv_trig_t VEML6075::getTrigger(void)
     {
         return TRIGGER_INVALID;
     }
-    return (conf & VEML6075_TRIG_MASK) >> VEML6075_TRIG_SHIFT;    
+    return static_cast<VEML6075::veml6075_uv_trig_t>((conf & VEML6075_TRIG_MASK) >> VEML6075_TRIG_SHIFT);
 }
 
 
@@ -310,15 +312,15 @@ VEML6075::veml6075_af_t VEML6075::getAutoForce(void)
     {
         return AF_INVALID;
     }
-    return (conf & VEML6075_AF_MASK) >> VEML6075_AF_SHIFT;        
+    return static_cast<VEML6075::veml6075_af_t>((conf & VEML6075_AF_MASK) >> VEML6075_AF_SHIFT);
 }
 
-VEML6075_error_t VEML6075::powerOn(boolean enable = true)
+VEML6075_error_t VEML6075::powerOn(boolean enable)
 {
     return shutdown(!enable);
 }
 
-VEML6075_error_t VEML6075::shutdown(boolean shutdown = true)
+VEML6075_error_t VEML6075::shutdown(boolean shutdown)
 {
     VEML6075_error_t err;
     veml6075_t conf;

